grai-integrations/source-snowflake/src/grai_source_snowflake/loader.py

```diff
@@ -0,0 +1,188 @@
+            self._connection = snowflake.connector.connect(**self.connection_dict)
+        return self
+
+    @property
+    def connection(self):
+        if self._connection is None:
+            raise Exception("Not connected, call `.connect()")
+        return self._connection
+
+    def close(self) -> None:
+        self.connection.close()
+        self._connection = None
+
+    def query_runner(self, query: str, param_dict: Dict = {}) -> List[Dict]:
+        dict_cursor = self.connection.cursor(snowflake.connector.DictCursor)
+        dict_cursor.execute(query, param_dict)
+        result = dict_cursor.fetchall()
+        #return [dict(item) for item in result]
+        return result
+
+    def get_tables(self) -> List[Table]:
+        """
+        Create and return a list of dictionaries with the
+        schemas and names of tables in the database
+        connected to by the connection argument.
+        """
+
+        query = """
+	        SELECT table_schema, table_name, table_type
+            FROM information_schema.tables
+            WHERE table_schema != 'INFORMATION_SCHEMA'
+            ORDER BY table_schema, table_name
+        """
+        res = ({k.lower(): v for k, v in result.items()} for result in self.query_runner(query))
+
+        addtl_args = {
+            "namespace": self.namespace,
+            "table_database": self.connection_dict['database']
+        }
+        print('table run')
+
+        return [Table(**result, **addtl_args) for result in res]
+
+    def get_columns(self, table: Table) -> List[Column]:
+        """
+        Creates and returns a list of dictionaries for the specified
+        schema.table in the database connected to.
+        """
+
+        query = f"""
+            SELECT column_name, data_type, is_nullable, column_default
+            FROM information_schema.columns
+            WHERE table_schema = '{table.table_schema}'
+            AND table_name = '{table.name}'
+        """
+
+        res = ({k.lower(): v for k, v in result.items()} for result in self.query_runner(query))
+
+        addtl_args = {
+            "namespace": table.namespace,
+            "schema": table.table_schema,
+            "table": table.name,
+        }
+        print('column run')
+        return [Column(**result, **addtl_args) for result in res]
+
+    def get_foreign_keys(self) -> List[Edge]:
+        """This needs to be tested / evaluated
+        :param connection:
+        :param table:
+        :return:
+        """
+        # This query only returns foreign keys, there is also exported keys and primary keys. 
+        # Information schema itself doesn't carry the column for foreign keys
+        # show IMPORTED KEYS in database '{self.connection_dict['database']}'
+        query = f"""
+        show IMPORTED KEYS
+        """
+        addtl_args = {
+            "namespace": self.namespace,
+        }
+
+        res = self.query_runner(query)
+
+        for i in res:
```

Comment:
This is me doing a little bike shedding here so feel free to ignore :) but I'd suggest using more descriptive variable names here (`i`, `y`, `z` would generally only be used when iterating over say an integer sequence like 1, 2, 3, ...)

If there is a descriptive name for the output of the query that would be better. I'm not sure what would be best here but something like 

```
imported_keys = self.query_runner(query)

for item in imported_keys:
...
```

would be an improvement